src/game/common/audio/audiomanager.cpp

```diff
@@ -709,17 +709,21 @@ void AudioManager::Set_Audio_Event_Volume_Override(Utf8String event, float vol_o
         Adjust_Volume_Of_Playing_Audio(event, vol_override);
     }
 
-    if (m_eventVolumeList.empty() && vol_override != -1.0f) {
-        m_eventVolumeList.push_back(std::pair<Utf8String, float>(event, vol_override));
-    } else {
-        for (auto it = m_eventVolumeList.begin(); it != m_eventVolumeList.end(); ++it) {
-            if (event == it->first && vol_override != -1.0f) {
-                it->second = vol_override;
+    for (auto it = m_eventVolumeList.begin(); it != m_eventVolumeList.end(); ++it) {
```

Comment:
could make this:
```cpp
auto res = std::find_if(m_eventVolumeList.begin(), m_eventVolumeList.end(), [&event](std::pair<Utf8String, float> item){
   return event == item->first;
}
if (res != m_eventVolumeList.end()){
    if (vol_override == -1.0f){
        m_eventVolumeList.erase(res);
    }
    else {
        res->second = vol_override;
    }
}
else
{
    if(vol_override != -1.0f){
        m_eventVolumeList.push_front(std::pair<Utf8String, float>(event, vol_override));
    }
}
```
Not sure if it reads better or not makes it perhaps a bit clearer as to what all the if's represent.